### PR TITLE
core api: add change-user-password

### PR DESCRIPTION
**Pull request by gsanchietti**
The UI can now change the password on first run

### CODE REVIEW

File: core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/50update [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import json
+import agent
+import redis
+
+request = json.load(sys.stdin)
+
+# Try to authenticate with user and password using unprivileged connection
+auth_db = redis.Redis(host='localhost', db=0, port=6379)
+if auth_db.execute_command('AUTH', request['user'], request['current_password']):
+    # On auhtentication success, switch to privileged connection to chang the password
```
Comment (DavidePrincipi) [inline core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/50update:33]:
```suggestion
    # On authentication success, switch to privileged connection to change the password
```

File: core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/validate-input.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    "new_password": {
+      "description": "New password in plain text. The password will be updated only the authentication with 'user' and 'current_password' si valid.",
+      "type": "string"
+    }
+  },
+  "required": [
+    "user",
+    "current_password",
+    "new_password"
```
Comment (DavidePrincipi) [inline core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/validate-input.json:31]:
```suggestion
    "new_password",
  ]
```

File: core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/validate-input.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "user": {
+      "description": "Target user.",
+      "type": "string"
+    },
+    "current_password": {
+      "description": "Current password in plain text",
+      "type": "string"
+    },
+    "new_password": {
+      "description": "New password in plain text. The password will be updated only the authentication with 'user' and 'current_password' si valid.",
```
Comment (DavidePrincipi) [inline core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/validate-input.json:24]:
```suggestion
      "description": "New password in plain text. The password will be updated only if the authentication with 'user' and 'current_password' is valid.",
```

File: core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/50update [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    rdb = agent.redis_connect(privileged=True)
+    rdb.acl_setuser(request['user'], passwords=[f'+{request["new_password"]}'], reset_passwords=True, enabled=True)
+    json.dump({}, fp=sys.stdout)
```
Comment (DavidePrincipi) [inline core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/50update:36]: I'd prefer returning a specific error code (and message). If the AUTH command fails I'd expect the UI behaves like a validation error occurs.

```suggestion
    json.dump({}, fp=sys.stdout)
else:
   # XXX validation error: invalid password
   sys.exit(2)
```

File: core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/validate-input.json (hunk 2 above)
Comment (gsanchietti) [reply core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/validate-input.json:31]: The comma is not required.

File: core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/50update (hunk 4 above)
Comment (gsanchietti) [reply core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/50update:36]: Let's ask @andre8244

Comment (DavidePrincipi) [reply core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/50update:36]: Ok, but a step abort with a sys.exit(...) line is needed in any case. The current code seems to fail only due to a output validation error, isn't it?

Comment (gsanchietti) [reply core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/50update:36]: @andre8244 prefers a validation error, let's go with it.

> The current code seems to fail only due to a output validation error, isn't it?

The code will raise an exception and exit with 1 if `acl_setuser` fails

Comment (DavidePrincipi) [reply core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/50update:36]: Ok, that's a runtime error case. The point is the AUTH failure: if it fails that's a user-input error, thus I think it has to be handled like a validation error.

File: core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/validate-input.json (hunk 2 above)
Comment (DavidePrincipi) [reply core/imageroot/var/lib/nethserver/cluster/actions/change-user-password/validate-input.json:31]: Right, the square bracket is required, though!

**Review by DavidePrincipi — APPROVED** [review]

**Pull request merged**